docs/docs/migration-guide.mdx

```diff
@@ -10,6 +10,63 @@ description: |
 This page contains information about changes between major versions and
 how you can migrate from one version to another.
 
+## Rasa 2.x to 3.0
+
+### Training data
+
+`TrainingDataImporter` and all its implementations are updated to contain only synchronous methods.
+If you have a custom data importer or rely on some functions provided by `TrainingDataImporter`, you need
+to update your implementation and function calls.
+
+For example, this is how the data loading should look like in Rasa 3.0:
+
+```python
+from typing import Text
+from rasa.shared.importers.importer import TrainingDataImporter
+
+def load_data(domain_path: Text, config_path: Text):
+    file_importer = TrainingDataImporter.load_from_config(
+        config_path, domain_path
+    )
+    # note that all the functions below were async before:
+    config = file_importer.get_config()
+    domain = file_importer.get_domain()
+    stories = file_importer.get_stories()
+    nlu_data = file_importer.get_nlu_data()
+```
+
+Since any custom importer implements `TrainingDataImporter`, you should update your custom
+importer to contain only sync methods as well:
+
+```python
+from typing import Dict, Text
+
+from rasa.shared.core.domain import Domain
+from rasa.shared.importers.importer import TrainingDataImporter
+
+
+class MyImporter(TrainingDataImporter):
+    """Example partial implementation of a custom importer component."""
+
+    # this function was async before
+    def get_domain(self) -> Domain:
+        path_to_domain_file = self._custom_get_domain_file()
+        return Domain.load(path_to_domain_file)
+
+    def _custom_get_domain_file(self) -> Text:
+        pass
```

Comment:
optional: how about shortening that part as it's not of interest for this example?

```suggestion
        pass

    def _custom_get_domain_file(self) -> Text:
        pass
```